### ISSUE TITLE
Inconsistent camera set across the three rasterization scripts (3 views in ego/perturbed vs 4 views in aug)

### ISSUE THREAD

**Issue by JoyDeng1004** [vita-epfl/RAP#20]
Hi, and thank you for releasing RAP! I think it’s a really elegant way to avoid the cost of generating synthetic data for E2E planning.

While reading the preprocessing code, I noticed an asymmetry across the three metadata scripts that I could not explain from the paper, and I would appreciate clarification before I build on top of it.

`ScenarioRenderer` defaults to **three** camera channels ([`process_data/helpers/renderer.py#L695`](https://github.com/vita-epfl/RAP/blob/main/process_data/helpers/renderer.py#L695)):

```python
def __init__(
    self,
    camera_channel_list=['CAM_F0', 'CAM_L0', 'CAM_R0'],
    width=1920,
    height=1120,
    ...
)
```

The three metadata scripts instantiate it differently:

| Script | Line | Camera channels rendered |
| --- | --- | --- |
| `create_openscene_metadata.py` | [L218](https://github.com/vita-epfl/RAP/blob/main/process_data/create_openscene_metadata.py#L218) | `ScenarioRenderer()` → **3** (`CAM_F0`, `CAM_L0`, `CAM_R0`) |
| `create_openscene_metadata_purturbed.py` | [L293](https://github.com/vita-epfl/RAP/blob/main/process_data/create_openscene_metadata_purturbed.py#L293) | `ScenarioRenderer()` → **3** (`CAM_F0`, `CAM_L0`, `CAM_R0`) |
| `create_openscene_metadata_aug.py` | [L214](https://github.com/vita-epfl/RAP/blob/main/process_data/create_openscene_metadata_aug.py#L214) | `ScenarioRenderer(camera_channel_list=['CAM_F0', 'CAM_L0', 'CAM_R0', 'CAM_B0'])` → **4** |

Meanwhile, the model side consistently expects **four** views. The consequence appears to be that, for the `rap_ego` and `rap_perturbed` caches, the `CAM_B0` slot of `rendered_camera_feature` is a **constant image** (zeros before normalization), while the corresponding real `CAM_B0` image is loaded normally.

## Questions

**1. Why do the three scripts render different camera sets?**

Is the 3-channel default in `create_openscene_metadata.py` and `create_openscene_metadata_purturbed.py` intentional? Was CAM_B0 left out to save rendering time, or because the rear view was found to contribute little?

**2. Given that only 3 channels are rendered for ego/perturbed, why is the real `CAM_B0` image still loaded and aligned against a constant raster?**

`RAPAgent.get_sensor_config()` requests `cam_b0` alongside the other three ([`rap_agent.py#L126-L134`](https://github.com/vita-epfl/RAP/blob/main/navsim/agents/rap_dino/rap_agent.py#L126)), so the real rear-view image is loaded and encoded normally. On the raster side, that same slot is the zero fallback. The alignment term then runs unmasked over the full view dimension ([`agent_lightning_module.py#L246-L250`](https://github.com/vita-epfl/RAP/blob/main/navsim/planning/training/agent_lightning_module.py#L246)):

```python
render_bev = bev_feature[:batch_size][real_valid_mask].detach()   # (N, 4, num_patch, C)
real_bev   = bev_feature[batch_size:]
loss += distill_feature_weight * F.mse_loss(render_bev, real_bev)
```

Since `F.mse_loss` averages over all elements and the raster side is detached, roughly a quarter of this term pulls the real rear-view features toward a constant target.

Was a per-view mask considered here, so that only the three views with an actual raster are aligned? I would also like to know whether the current behavior is intentional, for example because it was found to be negligible at `distill_feature_weight = 0.002` or useful as a form of regularization, or whether it is simply an unintended consequence of the silent zero fallback.

Any clarification on the above would be much appreciated.
